syft/pkg/cataloger/javascript/cataloger.go

```diff
@@ -20,3 +20,13 @@ func NewLockCataloger() *generic.Cataloger {
 		WithParserByGlobs(parseYarnLock, "**/yarn.lock").
 		WithParserByGlobs(parsePnpmLock, "**/pnpm-lock.yaml")
 }
+
+// NewJavaScriptCataloger returns a new JavaScript cataloger object based on detection
+// of npm based packages and lock files to provide a complete dependency graph of the
+// packages.
+func NewJavaScriptCataloger() *generic.GroupedCataloger {
+	return generic.NewGroupedCataloger("javascript-cataloger").
+		WithParserByGlobColocation(parseJavaScript, "**/yarn.lock", []string{"**/package.json", "**/yarn.lock"}).
```

Comment:
I think there is a fundamental issue with the cataloger changes here. You can specify a set of primary globs and a secondary set of required globs, but there isn't a way to say that the secondary results must come from a sibling path relative to the primary path. Instead the secondary match can be written relative to the root of the source.

In this case you could have a yarn.lock in a subdirectory `/opt/place/yarn.lock` and an unrelated javascript project in `/home/myuser/code/` that container `package.json` and `/yarn.lock` files... I think this cataloger would relate those findings together, mixing these two dependency trees together (which is not desirable)

Reply:
This should return a groupedParser for each of those locations

One for the yarn.lock and package.json at /opt/place/ and one for the package.json and yarn.lock at /home/myuser/code.

We generate a list of colocated files with `collectColocatedFiles` and then we determine if we have both the primary and secondary file in `isAllGlobsMatched`

Reply:
ahh, I see now -- thanks for the clarification 👍